autoload/lsp/omni.vim

```diff
@@ -63,14 +63,24 @@ function! lsp#omni#complete(findstart, base) abort
             while s:completion['status'] is# s:completion_status_pending && !complete_check()
                 sleep 10m
             endwhile
-            let l:base = tolower(a:base)
-            let s:completion['matches'] = filter(s:completion['matches'], {_, match -> stridx(tolower(match['word']), l:base) == 0})
+            let s:completion['matches'] = filter(s:completion['matches'], {_, match -> s:prefix_match(match['word'], a:base)})
```

Comment:
Now that I think of this seems like it would be good if we can optimize this by caching the tolower for prefix.